exec/cnex/nstring.c

```diff
@@ -145,6 +179,17 @@ void string_resizeString(TString *s, size_t n)
     s->length = n;
 }
 
+TString *string_reverseString(TString *s, size_t start, size_t end)
+{
+    TString *r = string_createString(end-start);
+    for (size_t i = start, e = end; e > start; e--, i++) {
+        r->data[e] = s->data[i];
+    }
+    s->data = r->data;
+    string_freeString(r);
```

Comment:
Does this do the right thing here? First you assign the `data` pointer from `r` to `s`, and then you free `r`. Doesn't that also free `r->data`, which is also pointed to by `s->data`? Leaving `s->data` pointing to freed memory?

Reply:
It surely did not.  Fixed this to use string_copyString().